src/sbml/AlgebraicRule.cpp

```diff
@@ -371,7 +371,7 @@ AlgebraicRule_create(unsigned int level, unsigned int version)
     AlgebraicRule* obj = new AlgebraicRule(level,version);
     return obj;
   }
-  catch (SBMLConstructorException)
+  catch (SBMLConstructorException &e)
```

Comment:
@skeating instead of defining a variable that we don't use, would it not be better to just change the type to `SBMLConstructorException &` without defining the variable?

Reply:
Variables you don't use do produce warnings, and the fewer of those the better. But wouldn't passing on the error message somehow be better still?

Reply:
in this case, here it is a c-function, that is trying to create something with invalid arguments, so returning `NULL` would be the expected result. I certainly wouldn't want to have messages printed anywhere. 

In the most cases where this is caught it is from internal ::createXXX functions, that take level / version from the SBMLDocument. So something would already have been very wrong long before the create call for this exception to ever be raised.